core/jvm/src/main/scala/zio/internal/RingBuffer.scala

```diff
@@ -49,111 +50,101 @@ object RingBuffer {
 }
 
 /**
- * A lock-free array-based bounded queue. It is thread-safe and can be
- * used in multiple-producer/multiple-consumer (MPMC) setting.
+ * A lock-free array-based bounded queue. It is thread-safe and can be used in
+ * multiple-producer/multiple-consumer (MPMC) setting.
  *
  * =Main concepts=
  *
- * A simple array-based queue of size N uses an array `buf` of size N
- * as an underlying storage. There are 2 pointers `head` and
- * `tail`. The element is enqueued into `buf` at position `tail % N`
- * and dequeued from `head % N`. Each time an enqueue happens `tail`
- * is incremented, similarly when dequeue happens `head` is
- * incremented.
+ * A simple array-based queue of size N uses an array `buf` of size N as an
+ * underlying storage. There are 2 pointers `head` and `tail`. The element is
+ * enqueued into `buf` at position `tail % N` and dequeued from `head % N`. Each
+ * time an enqueue happens `tail` is incremented, similarly when dequeue happens
+ * `head` is incremented.
  *
- * Since pointers wrap around the array as they get incremented such
- * data structure is also called a
- * [[https://en.wikipedia.org/wiki/Circular_buffer circular buffer]]
- * or a ring buffer.
+ * Since pointers wrap around the array as they get incremented such data
+ * structure is also called a
+ * [[https://en.wikipedia.org/wiki/Circular_buffer circular buffer]] or a ring
+ * buffer.
  *
- * Because queue is bounded, enqueue and dequeue may fail, which is
- * captured in the semantics of `offer` and `poll` methods.
+ * Because queue is bounded, enqueue and dequeue may fail, which is captured in
+ * the semantics of `offer` and `poll` methods.
  *
- * Using `offer` as an example, the algorithm can be broken down
- * roughly into three steps:
- *  1. Find a place to insert an element.
- *  2. Reserve this place, put an element and make it visible to
- *     other threads (store and publish).
- *  3. If there was no place on step 1 return false, otherwise
- *     returns true.
+ * Using `offer` as an example, the algorithm can be broken down roughly into
+ * three steps:
+ *   1. Find a place to insert an element. 2. Reserve this place, put an element
+ *      and make it visible to other threads (store and publish). 3. If there
+ *      was no place on step 1 return false, otherwise returns true.
```

Comment:
Newlines will be required in these sections.

Reply:
Overall, I still think this will be worth it. For overall consistency and saved manual formatting-time.

Reply:
Aha! Apparently for numbered lists to render correctly, you are supposed to use the following syntax:

<img width="664" alt="CleanShot 2021-11-07 at 12 43 02@2x" src="https://user-images.githubusercontent.com/7587245/140655805-eb1baa51-6455-42ce-9351-e2f2c9c08098.png">

Then it will render like this (it will also be auto-formatted correctly):

<img width="695" alt="CleanShot 2021-11-07 at 12 42 59@2x" src="https://user-images.githubusercontent.com/7587245/140655810-c5cc2595-300b-4acb-967c-6c6836242c77.png">